devices/Max1704x/Max1704X.cs

```diff
@@ -0,0 +1,314 @@
+        /// sequence is exceedingly noisy such that excess error is introduced into the
+        /// IC’s “first guess” of SOC, the host can issue a quick-start to reduce the
+        /// error. A quick-start is initiated by a rising edge on the QSTRT pin, or
+        /// through software by writing 4000h to MODE register.
+        /// Note: on the MAX17048/49 this will also clear / disable EnSleep.
+        /// </summary>
+        public void QuickStart()
+        {
+            Write16((ushort)Registers16.Max17043ModeQuickstart, (byte)Registers.Max17043Mode);
+        }
+
+        /// <summary>
+        /// Reset the device to its initial state by writing a specific value to the CMD Register.
+        /// This causes the device to reset as if power had been removed, and the reset occurs when
+        /// the last bit of the value has been clocked in.
+        /// </summary>
+        public void Reset()
+        {
+            Write16((ushort)Registers16.Max17043CommandPor, (byte)(byte)Registers.Max17043Command);
+        }
+
+        /// <summary>
+        /// Negates a byte value by performing a bitwise complement operation.
+        /// </summary>
+        /// <param name="value">The byte value to negate.</param>
+        /// <returns>The negation of the byte value.</returns>
+        protected static byte Negate(byte value)
+        {
+            return (byte)~value;
+        }
+
+        /// <summary>
+        /// Negates the given unsigned short value by performing a bitwise NOT operation.
+        /// </summary>
+        /// <param name="value">The unsigned short value to be negated.</param>
+        /// <returns>The result of the bitwise NOT operation on the value.</returns>
+        protected static ushort Negate(ushort value)
+        {
+            return (ushort)~value;
+        }
+
+        /// <summary>
+        /// Negates the given integer value.
+        /// </summary>
+        /// <param name="value">The value to be negated.</param>
+        /// <returns>The negated value as an unsigned 16-bit integer.</returns>
+        protected static ushort Negate(int value)
+        {
+            return (ushort)~value;
+        }
+        
+        /// <summary>
+        /// Retrieves the value of the config register from the MAX17043 register.
+        /// </summary>
+        /// <returns>The config register value as an unsigned short.</returns>
+        protected ushort GetConfigRegister()
+        {
+            return Read16((byte)Registers.Max17043Config);
+        }
+
+        /// <summary>
+        /// Writes a 16-bit value to the specified I2C address.
+        /// </summary>
+        /// <param name="data">The 16-bit value to write.</param>
+        /// <param name="address">The register address.</param>
+        /// <exception cref="Exception">Thrown when the write operation fails to complete.</exception>
+        protected void Write16(ushort data, byte address)
```

Comment:
same with the nice enum for address. The data is fine as ushort as you write multiple things here, not only the Registers16.

Reply:
done